lti_consumer/lti_1p3/extensions/rest_framework/serializers.py

```diff
@@ -89,3 +90,159 @@ class Meta:
+        if not super().is_valid(raise_exception=raise_exception):
+            return False
+
+        # `self.validated_data` is only set if `super().is_valid()` succeeds
+        new_instance = self.Meta.model(**self.validated_data)
+        try:
+            # Validate the instance
+            new_instance.full_clean()
+        except serializers.DjangoValidationError as exc:
+            errors = {key: value for key, value in exc.message_dict.items() if key != 'line_item'}
+            if errors:
+                # Unset any existing validated data because the model failed to validate
+                self._validated_data = {}  # pylint: disable=attribute-defined-outside-init
+
+                if raise_exception:
+                    raise serializers.ValidationError(errors)
+
+                return False
+
+        return True
+
+    class Meta:
+        model = LtiAgsScore
+        fields = (
+            'timestamp',
+            'scoreGiven',
+            'scoreMaximum',
+            'comment',
+            'activityProgress',
+            'gradingProgress',
+            'userId',
+        )
+
+
+class LtiAgsResultSerializer(serializers.ModelSerializer):
+    """
+    LTI AGS LineItemResult Serializer.
+
+    This maps out the internally stored LtiAgsScpre to
+    the LTI-AGS API Specification, as shown in the example
+    response below:
+
+    {
+      "id": "https://lms.example.com/context/2923/lineitems/1/results/5323497",
+      "scoreOf": "https://lms.example.com/context/2923/lineitems/1",
+      "userId": "5323497",
+      "resultScore": 0.83,
+      "resultMaximum": 1,
+      "comment": "This is exceptional work."
+    }
+
+    Reference:
+    https://www.imsglobal.org/spec/lti-ags/v2p0#example-application-vnd-ims-lis-v1-score-json-representation
+    """
+
+    id = serializers.SerializerMethodField()
+    scoreOf = serializers.SerializerMethodField()
+    userId = serializers.CharField(source='user_id')
+    resultScore = serializers.FloatField(source='score_given')
+    resultMaximum = serializers.SerializerMethodField()
+    comment = serializers.CharField()
+
+    def get_id(self, obj):
+        request = self.context.get('request')
+        return reverse(
+            'lti_consumer:lti-ags-view-results',
+            kwargs={
+                'lti_config_id': obj.line_item.lti_configuration.id,
+                'pk': obj.line_item.pk,
+                'user_id': obj.user_id,
+            },
+            request=request,
+        )
+
+    def get_scoreOf(self, obj):
+        request = self.context.get('request')
+        return reverse(
+            'lti_consumer:lti-ags-view-detail',
+            kwargs={
+                'lti_config_id': obj.line_item.lti_configuration.id,
+                'pk': obj.line_item.pk
+            },
+            request=request,
+        )
+
+    def get_resultMaximum(self, obj):
+        if obj.score_maximum <= 0:
+            return 1
+
+        return obj.score_maximum
```

Comment:
@pcockwell We don't need this for now. You can remove this and the serializer method field above and just return the final value.
```suggestion

```

Reply:
@giovannicimolin I'm not quite sure I understand... Why are we removing this? It matches the spec right? The results will only ever pull scores with non-null values for `score_given`, and `score_maximum` is required when `score_given` is non-null. What do you mean this is not needed right now?

Reply:
@pcockwell You can leave this as is. I've misread the spec here, thanks for the heads up.